src/app/dashboard/playground/playground.component.ts

```diff
@@ -0,0 +1,181 @@
+import { AfterViewInit, Component, ElementRef, OnDestroy, OnInit, ViewChild } from '@angular/core';
+import Konva from 'konva';
+import { fromEvent, Subject } from 'rxjs';
+import { Hero } from 'src/app/core/interface/hero';
+import { GameService } from '../game.service';
+import { takeUntil } from 'rxjs/operators';
+import { cloneData } from 'src/app/core/utils/utils';
+import { MessageService } from 'src/app/message.service';
+
+@Component({
+  selector: 'app-playground',
+  templateUrl: './playground.component.html',
+  styleUrls: ['./playground.component.css']
+})
+export class PlaygroundComponent implements OnInit, AfterViewInit, OnDestroy {
+  @ViewChild('playground') playgroundRef!: ElementRef;
+  stage: Konva.Stage | undefined;
+  layer: Konva.Layer | undefined;
+  layerBg: Konva.Layer | undefined;
+  heroesPlaying: Hero[] = [];
+  imageSize = {
+    width: 0,
+    height: 0
+  };
+  $unSubscribe = new Subject();
+  numberCol = 5; // number hero in row
+  gameInterval: any;
+  isPlaying: boolean = false;
+
+  constructor(
+    private readonly gameService: GameService,
+    private readonly messageService: MessageService
+  ) { }
+
+  ngOnInit(): void {
+    this.gameService.updateGame
+    .pipe(takeUntil(this.$unSubscribe))
+    .subscribe(value => {
+      if (value) {
+        this.isPlaying = false;
+        clearInterval(this.gameInterval);
+      }
+    });
+
+    this.gameService.heroesPlaying
+    .pipe(takeUntil(this.$unSubscribe))
+    .subscribe((heroes: Hero[]) => {
+      if (this.playgroundRef) {
+        this.heroesPlaying = heroes;
+        if (this.gameService.drawAgain) {
+          this.drawHeroes(heroes);
```

Comment:
we likely don't need to draw all the heroes all over again.  If each conva object was subscribed to the Observable of the hero, weapon, armour then we can make changes on the layer level

Reply:
- When adding/removing hero, my goal is draw each hero from left to right in a row, and when remove a hero, heroes below will fill to position of hero removed. So I need  draw it again.
- After your comment, I will change it: I will add for each hero group an id. When adding/removing a hero, I will add/remove hero group by id. So I do not draw all heroes again when list change.